src/main/java/io/cdap/plugin/gcp/dataplex/source/DataplexBatchSource.java

```diff
@@ -0,0 +1,514 @@
+import com.google.cloud.storage.StorageException;
+import com.google.common.base.Strings;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.api.data.batch.Input;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.dataset.lib.KeyValue;
+import io.cdap.cdap.etl.api.Emitter;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.cdap.etl.api.PipelineConfigurer;
+import io.cdap.cdap.etl.api.StageConfigurer;
+import io.cdap.cdap.etl.api.batch.BatchRuntimeContext;
+import io.cdap.cdap.etl.api.batch.BatchSource;
+import io.cdap.cdap.etl.api.batch.BatchSourceContext;
+import io.cdap.cdap.etl.api.validation.ValidatingInputFormat;
+import io.cdap.plugin.common.LineageRecorder;
+import io.cdap.plugin.common.batch.JobUtils;
+import io.cdap.plugin.gcp.bigquery.source.BigQueryAvroToStructuredTransformer;
+import io.cdap.plugin.gcp.bigquery.source.BigQuerySourceUtils;
+import io.cdap.plugin.gcp.bigquery.util.BigQueryConstants;
+import io.cdap.plugin.gcp.bigquery.util.BigQueryUtil;
+import io.cdap.plugin.gcp.common.GCPConnectorConfig;
+import io.cdap.plugin.gcp.common.GCPUtils;
+import io.cdap.plugin.gcp.dataplex.common.util.DataplexConstants;
+import io.cdap.plugin.gcp.dataplex.common.util.DataplexUtil;
+import io.cdap.plugin.gcp.dataplex.source.config.DataplexBatchSourceConfig;
+import io.cdap.plugin.gcp.gcs.GCSPath;
+
+import org.apache.avro.generic.GenericData;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.lib.input.FileInputFormat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nullable;
+
+/**
+ * Batch Source that reads data from Dataplex entities (BigQuery or Cloud Storage).
+ * <p>
+ * {@code Object} is the first parameter because that is the key used
+ * by Hadoop's {@code TextInputFormat}.
+ * Object is the second parameter because that is the value used
+ * by Hadoop's {@code TextInputFormat}.
+ * {@code StructuredRecord} is the third parameter because that is the output given by
+ * Hadoop's {@code TextInputFormat}. All the plugins included with Hydrator operate on
+ * StructuredRecord.
+ */
+@Plugin(type = BatchSource.PLUGIN_TYPE)
+@Name(DataplexBatchSource.NAME)
+@Description("Dataplex Source")
+public class DataplexBatchSource extends BatchSource<Object, Object, StructuredRecord> {
+  public static final String NAME = "Dataplex";
+  private static final String BQ_TEMP_BUCKET_NAME_PREFIX = "bq-source-bucket-";
```

Comment:
nit: this can be confuses with the BigQuery source. Can we make this `dataplex-bq-source-bucket`?

Reply:
done